beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -644,8 +644,17 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
             "Reconstructed columns"
         );
 
+        let columns_to_sample = self
+            .custody_context()
+            .sampling_columns_for_epoch(slot.epoch(T::EthSpec::slots_per_epoch()), &self.spec);
+        let data_columns_to_import: Vec<_> = data_columns_to_publish
+            .iter()
+            .filter(|column| columns_to_sample.contains(&column.index()))
+            .cloned()
+            .collect();
```

Comment:
I made a spec PR here, as per our discussion this morning - I realise non supernondes would actually use more outbound bandwidth after reconstruction, i think we should just stick with publishing `data_columns_to_import`. It's a `SHOULD` on the current spec to publish all new columns.
https://github.com/ethereum/consensus-specs/pull/4657